README.md

```diff
@@ -125,6 +125,36 @@ GitHub CLI comes pre-installed in all [GitHub-Hosted Runners](https://docs.githu
 
 Download packaged binaries from the [releases page][].
 
+#### Verification of binaries
+
+Since version 2.50.0 `gh` has been producing [Build Provenance Attestation](https://github.blog/changelog/2024-06-25-artifact-attestations-is-generally-available/) enabling a cryptographically verifiable paper-trail back to the origin GitHub repository, git revision and build instructions used. The build provenance attestations are signed and relies on Public Good [Sigstore](https://www.sigstore.dev/) for PKI.
+
+There are two common ways to verify a downloaded release, depending if `gh` is aready installed or not. If `gh` is installed, it's trivial to verify a new release:
+
+```shell
+$ % gh at verify -R cli/cli gh_2.62.0_macOS_arm64.zip
+Loaded digest sha256:fdb77f31b8a6dd23c3fd858758d692a45f7fc76383e37d475bdcae038df92afc for file://gh_2.62.0_macOS_arm64.zip
+Loaded 1 attestation from GitHub API
+✓ Verification succeeded!
+
+sha256:fdb77f31b8a6dd23c3fd858758d692a45f7fc76383e37d475bdcae038df92afc was attested by:
+REPO     PREDICATE_TYPE                  WORKFLOW
+cli/cli  https://slsa.dev/provenance/v1  .github/workflows/deployment.yml@refs/heads/trunk
+```
+
+If `gh` is not installed, you can use the `cosign` [binary](https://github.com/sigstore/cosign) from Sigstore to verify a release. To perform this, you must first download the [attestation](https://github.com/cli/cli/attestations) for the downloaded release, then you can invoke cosign to verify the authenticity of the downloaded file:
+
+```shell
+$ cosign verify-blob-attestation --bundle cli-cli-attestation-3120304.sigstore.json \
+      --new-bundle-format \
+      --certificate-oidc-issuer="https://token.actions.githubusercontent.com" \
+      --certificate-identity-regexp="^https://github.com/cli/cli/.github/workflows/deployment.yml@refs/heads/trunk$" \
+      gh_2.62.0_macOS_arm64.zip
+Verified OK
+```
+
+This will verify that the downloaded zip file is authentic, and it can be extracted to get the `gh` binary.
+
```

Comment:
This is a little difficult for PR `suggestion` to deal with, but I think reformatting this will make it easier to read and recognize the 2 options:

> There are two common ways to verify a downloaded release, depending if `gh` is aready installed or not:
> 
> - **Option 1: Using `gh` if already installed:**
> 
>    ```shell
>    $ % gh at verify -R cli/cli gh_2.62.0_macOS_arm64.zip
>    Loaded digest sha256:fdb77f31b8a6dd23c3fd858758d692a45f7fc76383e37d475bdcae038df92afc for file://gh_2.62.0_macOS_arm64.zip
>    Loaded 1 attestation from GitHub API
>    ✓ Verification succeeded!
>    
>    sha256:fdb77f31b8a6dd23c3fd858758d692a45f7fc76383e37d475bdcae038df92afc was attested by:
>    REPO     PREDICATE_TYPE                  WORKFLOW
>    cli/cli  https://slsa.dev/provenance/v1  .github/workflows/deployment.yml@refs/heads/trunk
>    ```
> 
> - **Option 2: Using Sigstore [`cosign`](https://github.com/sigstore/cosign):**
> 
>    To perform this, download the [attestation](https://github.com/cli/cli/attestations) for the downloaded release and use cosign to verify the authenticity of the downloaded release:
>    
>    ```shell
>    $ cosign verify-blob-attestation --bundle cli-cli-attestation-3120304.sigstore.json \
>          --new-bundle-format \
>          --certificate-oidc-issuer="https://token.actions.githubusercontent.com" \
>          --certificate-identity-regexp="^https://github.com/cli/cli/.github/workflows/deployment.yml@refs/heads/trunk$" \
>          gh_2.62.0_macOS_arm64.zip
>    Verified OK
>    ```

Reply:
yeah, good idea. I'll rearrange the section 👍